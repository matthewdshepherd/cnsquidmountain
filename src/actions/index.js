export const addFetchedWords = fetchedData => ({
    type: 'ADD_FETCHED_WORDS',
    fetchedData
})

export const setUser = userName => ({
    type: 'SET_USER',
    userName
})