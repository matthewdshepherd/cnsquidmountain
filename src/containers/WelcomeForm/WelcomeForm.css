.welcomeForm__div {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-content: center;
    align-items: center;
    /* margin-top: 50%; */
    width: 100%;
}

.large_logo {
    height: 225px;
    width: 225px;
    margin-bottom: 50px;
}

.welcomeForm--title {
    font-size: 7em;
    margin: 20px 0;
    color: #FFF;
}

.welcomeForm {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    align-items: center;
    height: 200px;
}

.welcomeFormName__label {
    font-size: 23px;
    font-weight: 500;
    color: #FFF;
    font-size: 30px;
}

.welcomeFormName__input {
    border: none;
    height: 60px;
    font-size: 30px;
    text-align: center;
    font-weight: 500;
    border-radius: 10px;
    width: 100%;
}

::placeholder {
    font-weight: 300;
}

.welcomeFormName__button {
    height: 50px;
    border: none;
    border-radius: 10px;
    font-size: 30px;
    font-weight: 700;
    background-color: #CEBE41;
    color: #FFF;
}

.welcomeForm--welcome--text{
    font-size: 40px;
    font-weight: 300;
    margin: 0;
    align-self: center;
    color: orange;
    text-shadow: 2px 2px black;
}

.ready__div{
    display: flex;
    justify-content: space-between;
}

.welcomeForm--ready--text{
    font-size: 30px;
    font-weight: 300;
    margin: 0;
    align-self: center;
    text-shadow: 1px 1px white;
}

.currentPlayer--name {
    font-size: 70px;
    font-weight: 700;
    color: #FFF;
}

.display-none {
    display: none;
}

.start__button--wrapper {
    display: block;
    width: 30%;
    height: 40px;
}

.start__button--border {
    position: relative;
    height: 40px;
    border: 2px solid #89849b;
    border-radius: 10px;
}

.welcomeForm--start--button {
    position: absolute;
    width: 100%;
    height: 40px;
    cursor: pointer;
}

.start__button--pulse {
    height: 40px;
    width: 100%;
    position: absolute;
    margin: 0;
    animation: pulsate 2s ease-out;
    animation-iteration-count: infinite;
    opacity: 0.0;
    box-shadow: 0 0 1px 2px orange;
    animation-delay: 2s;
    border-radius: 10px;
}


@keyframes pulsate {
    0% {
      transform: scale(0.9, 0.9);
      opacity: 0.0;
    }
    50% {
      opacity: 1.0;
    }
    100% {
      transform: scale(1.3, 1.3);
      opacity: 0;
    }
  }