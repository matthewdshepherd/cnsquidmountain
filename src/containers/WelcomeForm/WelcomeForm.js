import React from 'react'
import { bindActionCreators } from 'redux';
import { Link } from "react-router-dom";
import { connect } from 'react-redux'
import { setPlayer, ramdomizeGameData } from '../../actions'
import './WelcomeForm.css'

import squidMountainLogo from '../../images/squid_mountain_logo.png'

export class WelcomeForm extends React.Component {
    constructor(){
        super()
        this.state = {
            name: ''
        };
    };
    
    handleChange = event => {
        this.setState({ [event.target.name]: event.target.value })
    }
    
    handleSubmit = event => {
        event.preventDefault();
        const { setPlayer } = this.props
        setPlayer(this.state.name)
        this.setState({
            name: ''
        })
    }

    canBeSubmitted() {
        const { name } = this.state
        return name.length > 0
    }

    handleStart = event => {
        console.log(this.props)
        const { ramdomizeGameData,  gameData} = this.props
        ramdomizeGameData(gameData)
    }


    render() {
        const isEnabled =  this.canBeSubmitted();
        return(
            <div className="welcomeForm__div">
                <h1 className="welcomeForm--title">Squid Mountain</h1>
                <img
                src={squidMountainLogo}
                alt="Squid Mountain Logo"
                className="large_logo"
                />
                <form className="welcomeForm">
                    <label htmlFor="name" className={this.props.currentPlayer ? "display-none" : "welcomeFormName__label"}>Mountaineer's Name:</label>
                    <input className={this.props.currentPlayer ? "display-none" : "welcomeFormName__input"} id="name" type="text" name="name" placeholder="Enter Name Here" onChange={this.handleChange} value={this.state.name} />
                    <button  type="button" className={this.props.currentPlayer ? "display-none" : "welcomeFormName__button"} disabled={!isEnabled} onClick={this.handleSubmit}>Enter</button>
                    <p className={this.props.currentPlayer ? "welcomeForm--welcome--text" : "display-none"}>Welcome,</p>
                    <p className={this.props.currentPlayer ? "currentPlayer--name" : "display-none"}>{this.props.currentPlayer}!</p>
                    <div className="ready__div">
                        <p className={this.props.currentPlayer ? "welcomeForm--ready--text" : "display-none"}>Ready to go?</p>
                        <div className={ this.props.currentPlayer ? "start__button--wrapper" : "display-none"} >
                            <div className="start__button--border">
                                <div className="start__button--pulse" ></div>
                                <Link to='/play'>
                                    <button
                                    type='button'
                                    className="welcomeForm--start--button"
                                    onClick={this.handleStart}
                                    >Start</button> 
                                </Link>
                            </div>
                        </div>
                    </div>
                </form>
            </div>
        )
    }
};

const matStateToProps = ({ currentPlayer, gameData }) => ({ currentPlayer, gameData})

const mapDispatchToProps = dispatch => (bindActionCreators ({
    setPlayer,
    ramdomizeGameData
}, dispatch))

export default connect(matStateToProps, mapDispatchToProps)(WelcomeForm);