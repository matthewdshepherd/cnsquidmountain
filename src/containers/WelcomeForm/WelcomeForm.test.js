import React from 'react'
import { shallow } from 'enzyme'
import { WelcomeForm, mapStateToProps, mapDispatchToProps } from './WelcomeForm'
import { setPlayer, ramdomizeGameData } from '../../actions/index'

jest.mock('../../Utilities/apiCalls')

describe('Welcome', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(<WelcomeForm />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})

describe('mapDispatchToProps', () => {
  it('calls dispatch with setPlayer', () => {
    const mockDispatch = jest.fn()
    const mockaction = jest.fn()
    const wrapper = shallow(<WelcomeForm setPlayer={mockaction} />)
    const mockFetchedData = 'Brittney Spears'
    const actionToDispatch = setPlayer('SET_PLAYER', mockFetchedData)

    const mappedProps = mapDispatchToProps(mockDispatch)

    mappedProps.setPlayer('SET_PLAYER', mockFetchedData)

    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
  })
})
