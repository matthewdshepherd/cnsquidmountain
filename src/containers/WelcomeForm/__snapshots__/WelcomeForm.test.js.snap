// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome should match the snapshot 1`] = `
<div
  className="welcomeForm__div"
>
  <h1
    className="welcomeForm--title"
  >
    Squid Mountain
  </h1>
  <img
    alt="Squid Mountain Logo"
    className="large_logo"
    src="squid_mountain_logo.png"
  />
  <form
    className="welcomeForm"
  >
    <label
      className="welcomeFormName__label"
      htmlFor="name"
    >
      Mountaineer's Name:
    </label>
    <input
      className="welcomeFormName__input"
      id="name"
      name="name"
      onChange={[Function]}
      placeholder="Enter Name Here"
      type="text"
      value=""
    />
    <button
      className="welcomeFormName__button"
      disabled={true}
      onClick={[Function]}
      type="button"
    >
      Enter
    </button>
    <p
      className="display-none"
    >
      Welcome,
    </p>
    <p
      className="display-none"
    >
      !
    </p>
    <div
      className="ready__div"
    >
      <p
        className="display-none"
      >
        Ready to go?
      </p>
      <div
        className="display-none"
      >
        <div
          className="start__button--border"
        >
          <div
            className="start__button--pulse"
          />
          <Link
            to="/play"
          >
            <button
              className="welcomeForm--start--button"
              onClick={[Function]}
              type="button"
            >
              Start
            </button>
          </Link>
        </div>
      </div>
    </div>
  </form>
</div>
`;
